### ISSUE TITLE
How do I use it using mach-glfw?

### ISSUE THREAD

**Issue by GeezerLMAO** [ziglibs/zgl#64]
I know that you need to use the `loadExtension` function but I don't know how.

main.zig:

``` zig
const std = @import("std");
const glfw = @import("glfw");
const gl = @import("zgl");

pub fn main() !void {
    try glfw.init(.{});
    defer glfw.terminate();

    const window = try glfw.Window.create(1024, 576, "Hello World", null, null, .{});
    defer window.destroy();

    try glfw.makeContextCurrent(window);

    while (!window.shouldClose()) {
        try glfw.pollEvents();
        gl.clear(.{ .color = true, .depth = true, .stencil = false });
    }
}
```

Is there something missing?

**Comment by ikskuh** [comment]
You have to invoke the function `loadExtensions` (which is misnamed) to get all function calls loaded.

See https://github.com/ziglibs/zgl/blob/master/zgl.zig#L2206-L2208 and https://github.com/MasterQ32/zig-opengl#example

**Comment by GeezerLMAO** [comment]
I Tried to implement it and spits an error. I think my problem is that I don't know what to substitute `?gl.FunctionPointer` to


Error:
```
 zig build run
error(OpenGL): entry point glGetnTexImage not found!
error(OpenGL): entry point glGetnUniformdv not found!
error: EntryPointNotFound
main.zig:20:5: 0x7ff767cdb403 in main (TestGL.exe.obj)
    try gl.loadExtensions(proc, glGetProcAddress);
    ^
The following command exited with error code 1 (expected 0):
bin\TestGL.exe 
error: UnexpectedExitCode
C:\Zig-SDK\Zig\lib\std\build\RunStep.zig:277:17: 0x7ff7718da740 in runCommand (build.exe.obj)
                return error.UnexpectedExitCode;
                ^
C:\Zig-SDK\Zig\lib\std\build\RunStep.zig:183:5: 0x7ff7718baccb in make (build.exe.obj)
    try runCommand(
    ^
C:\Zig-SDK\Zig\lib\std\build.zig:3660:9: 0x7ff77185fd59 in make (build.exe.obj)
        try self.makeFn(self);
        ^
C:\Zig-SDK\Zig\lib\std\build.zig:509:9: 0x7ff77184c4cc in makeOneStep (build.exe.obj)
        try s.make();
        ^
C:\Zig-SDK\Zig\lib\std\build.zig:503:17: 0x7ff77184c445 in makeOneStep (build.exe.obj)
                return err;
                ^
C:\Zig-SDK\Zig\lib\std\build.zig:464:13: 0x7ff77184c198 in make (build.exe.obj)
            try self.makeOneStep(s);
            ^
C:\Zig-SDK\Zig\lib\build_runner.zig:223:21: 0x7ff77184f0f0 in main (build.exe.obj)
            else => return err,
```

Code:

``` zig
const std = @import("std");
const glfw = @import("glfw");
const gl = @import("zgl");

fn glGetProcAddress(p: glfw.GLProc, proc: [:0]const u8) ?*const anyopaque {
    _ = p;
    return glfw.getProcAddress(proc);
}

pub fn main() !void {
    try glfw.init(.{});
    defer glfw.terminate();

    const window = try glfw.Window.create(1024, 576, "Hello World", null, null, .{});
    defer window.destroy();

    try glfw.makeContextCurrent(window);

    const proc: glfw.GLProc = undefined;
    try gl.loadExtensions(proc, glGetProcAddress);

    while (!window.shouldClose()) {
        try glfw.pollEvents();
        gl.clear(.{ .color = true, .depth = true, .stencil = false });

        try window.swapBuffers();
    }
}
```

**Comment by ikskuh** [comment]
You have to make sure to have an OpenGL 4.5 Core context, otherwise those functions won't go away:
> ```
> zig build run
> error(OpenGL): entry point glGetnTexImage not found!
> error(OpenGL): entry point glGetnUniformdv not found!
>```

As you can see [here](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glGetTexImage.xhtml) and [here](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glGetUniform.xhtml), those functions are included in GL 4.5, but not earlier. `zgl` requires a 4.5 context

**Comment by heysokam** [comment]
@MasterQ32 This should be a basic example in the readme
I had to dig for 1h just to make it build correctly
Please add some compiling guidelines in a basic example somewhere
It feels unnecesary to have such friction for starting to use a library
